src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -99,6 +99,14 @@ def query_imported_fast(self):
         except libzfs.ZFSException as e:
             raise CallError(f'Failed listing imported pools with error: {e}')
 
+    def is_upgraded(self, pool_name):
+        is_upgraded = False
```

Comment:
We should probably raise `ENOENT` `CallError` here if the pool name was invalid and only keep `return False if the pool name is invalid` behavior for the public method.